src/crawlers/statistics.js

```diff
@@ -215,7 +215,9 @@ class Statistics {
         this.finishedJobs = savedState.finishedJobs;
         this.failedJobs = savedState.failedJobs;
         this.totalJobDurationMillis = savedState.totalJobDurationMillis;
-        this.startedAt = new Date(Date.now() - (savedState.persistedAt - savedState.startedAt));
+        // persisted state uses ISO date strings
+        const [persistedAt, startedAt] = ['persistedAt', 'startedAt'].map(key => new Date(savedState[key]).getTime());
+        this.startedAt = new Date(Date.now() - (persistedAt - startedAt));
```

Comment:
How about adding `this.runtimeSecs` or `this.runtimeMillis`. I missed this earlier, but it feels wrong to replace the `startedAt` on migration/restart. If we're keeping continuity of request numbers, we should also keep time continuity. The `startedAt` should always more or less respect the started at time on the platform, otherwise people would get confused. Also, if you later want to work with the persisted statistics to create some analytics, your times would be skewed.

Reply:
makes sense